webui/src/components/selectlanguage.jsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react/lib/ReactWithAddons';
+import { Link } from 'react-router'
+import { DropdownList } from 'react-widgets';
+import { serverCache, Error } from '../data/server';
+import { Wait, Err } from './waiting.jsx';
+
+export const SelectLanguage = React.createClass({
+    getInitialState: function(){
+        return{
+            value: '',
+            lastLangSearch: '',
+            langNum: 0,
+            maxLang: 10,
+        };
+    },
+
+    handleLangChange: function (val) {
+        this.setState({value: val.id});
+        this.props.onSelectLanguage(val.id);
+    },
+
+    filterLanguage: function (languageName, search){
+        // rendering all (>7000) languages is slow, so limit number of responses.
+
+        var ret = false;
+        if(search != this.state.lastLangSearch){
+            this.state.langNum = 0;
+        }
+
+        // search for either language name or iso 639-3 code
+        if((languageName.name.toLowerCase().indexOf(search.toLowerCase()) != -1 
+            || languageName.id.toLowerCase().indexOf(search.toLowerCase()) != -1)
+            && this.state.langNum < this.state.maxLang)
+        {
+            this.state.langNum = this.state.langNum+1;
+            ret = true
+        }
+        return ret;
+    },
+
+    componentWillMount() {
+        this.componentWillReceiveProps(this.props);
+    },
+
+    componentWillReceiveProps(props) {
+        var def = this.props.defValue || 'eng'; 
+        this.setState({value:def});
```

Comment:
@emanueldima There is a problem here that I couldn't fix. When the user choose a new value, the dropdown menu doesn't show it, while the parent is getting the new value. Can you please take a look?